scripts/feature_toggle_report_generator.py

```diff
@@ -0,0 +1,237 @@
+            template_data['creation_date'] = datetime.datetime.strftime(
+                self.data['created'], date_template
+            )
+            template_data['last_modified_date'] = datetime.datetime.strftime(
+                self.data['modified'], date_template
+            )
+        elif self.toggle_type == 'waffle.flag':
+            template_data['note'] = self.data['note']
+            template_data['creation_date'] = datetime.datetime.strftime(
+                self.data['created'], date_template
+            )
+            template_data['last_modified_date'] = datetime.datetime.strftime(
+                self.data['modified'], date_template
+            )
+            template_data['everyone'] = self.data['everyone']
+            template_data['percent'] = null_or_number(self.data['percent'])
+            template_data['testing'] = self.data['testing']
+            template_data['superusers'] = self.data['superusers']
+            template_data['staff'] = self.data['staff']
+            template_data['authenticated'] = self.data['authenticated']
+            template_data['languages'] = filter(
+                lambda x: x != '', self.data['languages'].split(',')
+            )
+            template_data['rollout'] = self.data['rollout']
+
+        return template_data
+
+
+class Renderer(object):
+
+    def __init__(self, template_dir, report_dir):
+        self.jinja_environment = jinja2.Environment(
+            autoescape=False,
+            loader=jinja2.FileSystemLoader('templates'),
+            lstrip_blocks=True,
+            trim_blocks=True
+        )
+        self.report_dir = report_dir
+        if not os.path.isdir(report_dir):
+            os.mkdir(report_dir)
+
+    def render_file(self, output_file_name, template_name, variables={}):
+        file_path = os.path.join(self.report_dir, output_file_name)
+        template = self.jinja_environment.get_template(template_name)
+        with io.open(file_path, 'w') as output:
+            output.write(
+                template.render(**variables)
+            )
+
+    def render_report(self, report_data):
+        """
+        Generate the entire feature toggle report
+        """
+        self.render_file(
+            'index.rst', 'index.tpl',
+            variables={'idas': report_data.keys()}
+        )
+        for ida_name, ida_data in report_data.items():
+            self.render_file(
+                '{}.rst'.format(ida_name), 'ida_base.tpl',
+                variables={'ida': ida_data}
+            )
+            self.render_file(
+                '{}-feature-toggle-state.rst'.format(ida_name), 'ida.tpl',
+                variables={'ida': ida_data}
+            )
+
+
+def add_toggle_state_to_idas(idas, dump_file_path):
+    """
+    Given a dictionary of Idas to consider, and the path to a directory
+    containing the SQL dumps for feature toggles in said Idas, read each dump
+    file, parsing and linking it's data into the Ida associated with it.
```

Comment:
nit: "it's" -> "its"

Reply:
Argh! I will never be able to get this one!